### PR TITLE
Captures search domain tap

### PR DESCRIPTION
**Pull request by antonis**
**Based on:** https://github.com/wordpress-mobile/WordPress-Android/pull/19485

## Description
This PR captures the tapped domain on the domains list so that it can be used on the next steps of the flow.

Note: This PR does not link up to the Purchase screen https://github.com/wordpress-mobile/WordPress-Android/pull/19332. The intention is to capture the `ProposedDomain` information so that it can be passed in the https://github.com/wordpress-mobile/WordPress-Android/issues/19382 implementation for testing adding the domain to the cart and opening the checkout webview.

## To test
1. Make sure you have enabled the domain_management feature flag.
2. Go to Me -> Domains
3. Tap on "+" button on the toolbar
4. Search for a domain
5. Tap on one of the domain results
6. **Verify** that the `🔵 Tracked: domain_management_search_domain_tapped, Properties: {"domain_name":"tapped.domain"}` tracking event is emitted 

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Added new test cases in `NewDomainSearchViewModelTest`

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/domains/management/newdomainsearch/composable/NewDomainSearchScreen.kt [hunk 1]
```diff
@@ -163,39 +166,50 @@ fun TransferDomainFooter(
...
-                text = domain.domainSuffix,
-                style = MaterialTheme.typography.bodyLarge,
-                fontWeight = FontWeight.Bold
-            )
-        }
-        if (domain.salePrice.isNullOrEmpty()) {
-            Text(text = stringResource(id = R.string.new_domain_search_screen_list_item_regular_price, domain.price))
-        } else {
+    Surface(onClick = { onDomainTapped(domain) }) {
+        Column(modifier = modifier.padding(16.dp)) {
```
Comment (justtwago) [inline WordPress/src/main/java/org/wordpress/android/ui/domains/management/newdomainsearch/composable/NewDomainSearchScreen.kt]:
```suggestion
        Column(modifier = Modifier.padding(16.dp)) {
```

File: WordPress/src/main/java/org/wordpress/android/ui/domains/management/newdomainsearch/composable/NewDomainSearchScreen.kt [hunk 2]
```diff
@@ -163,39 +166,50 @@ fun TransferDomainFooter(
...
-            Text(
-                text = domain.domainSuffix,
-                style = MaterialTheme.typography.bodyLarge,
-                fontWeight = FontWeight.Bold
-            )
-        }
-        if (domain.salePrice.isNullOrEmpty()) {
-            Text(text = stringResource(id = R.string.new_domain_search_screen_list_item_regular_price, domain.price))
-        } else {
+    Surface(onClick = { onDomainTapped(domain) }) {
```
Comment (justtwago) [inline WordPress/src/main/java/org/wordpress/android/ui/domains/management/newdomainsearch/composable/NewDomainSearchScreen.kt]: In that case `modifier` should be used in the Surface instead. In addition, I'd suggest to add `modifier.fillMaxWidth()` to make the entire row clickable.

```suggestion
    Surface(
        onClick = { onDomainTapped(domain) },
        modifier = modifier.fillMaxWidth()
    ) {
```

**Review by justtwago — COMMENTED** [review]
Thank you for making the results clickable. I left just a UI-related comment only.

**Comment by antonis** [comment]
> Note: This PR does not link up to the Purchase screen https://github.com/wordpress-mobile/WordPress-Android/pull/19332.

I've added a separate issue for this https://github.com/wordpress-mobile/WordPress-Android/issues/19502

**Comment by antonis** [comment]
> Thank you for making the results clickable. I left just a UI-related comment only.

Thank you for reviewing and the suggestions @justtwago 🙇 
I've committed your suggestions :)

**Review by justtwago — APPROVED** [review]
Thank you a lot for the changes! 🎉 Merging this PR.

**Pull request merged**